### PR TITLE
User Tracking cleanup

### PR DESCRIPTION
**Pull request by tippmar-nr**
Thank you for submitting a pull request.  Please review our [contributing guidelines](/CONTRIBUTING.md) and [code of conduct](https://opensource.newrelic.com/code-of-conduct/).

## Description

Minor cleanup to address requirements changes. Removed `userId` references from `ProcessNoticedError()`, added `IsNullOrEmpty()` check before attempting to set the userId in `CustomErrorDataTransformer.Transform()`, updated `EndUserId` to include `TransactionTrace` and `TransactionEvent` as destination types.

# Author Checklist
- [x] Unit tests, Integration tests, and Unbounded tests completed
- ~~[ ] Performance testing completed with satisfactory results (if required)~~
- ~~[ ] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated~~

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

### CODE REVIEW

**Comment by JcolemanNR** [comment]
This does not capture the requirements change fully. The user id needs to be included in transactions as an attribute, instead of a special field

**Review by JcolemanNR — CHANGES_REQUESTED** [review]
The user id still needs to be included in transactions and transaction events as an attribute.

File: src/Agent/NewRelic/Agent/Core/Transformers/CustomErrorDataTransformer.cs [hunk 1]
```diff
@@ -65,7 +65,10 @@ public void Transform(ErrorData errorData, float priority, string userid)
             // so it can display the stack trace. 
             _attribDefs.TransactionNameForError.TrySetValue(attribValues, errorData.Path);
 
-            _attribDefs.EndUserId.TrySetValue(attribValues, userid);
+            if (!string.IsNullOrEmpty(userid))
```
Comment (nrcventura) [inline src/Agent/NewRelic/Agent/Core/Transformers/CustomErrorDataTransformer.cs:68]: We might want to change it to something like the following.
```suggestion
            if (!string.IsNullOrWhitespace(userid))
```

Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Core/Transformers/CustomErrorDataTransformer.cs:68]: I prefer IsNullOrWhitespace as well.

Comment (tippmar-nr) [reply src/Agent/NewRelic/Agent/Core/Transformers/CustomErrorDataTransformer.cs]: Done. Plus in a couple of other places as well

**Review by nrcventura — APPROVED** [review]

**Comment by chynesNR** [comment]
Should we log something at Finest if a blank user ID is supplied? It might help troubleshoot "where's my user ID?" issues.

**Comment by nrcventura** [comment]
> Should we log something at Finest if a blank user ID is supplied? It might help troubleshoot "where's my user ID?" issues.

I don't know if the logger is available in the API when we first validate the user ID. I think that it's a helpful idea, but it might affect where we validate that input.

**Review by chynesNR — APPROVED** [review]

**Pull request merged**